portal-wire-protocol.md

```diff
@@ -0,0 +1,163 @@
+    * Individual ENR records **MUST** correspond to one of the requested distances.
+    * It is invalid to return multiple ENR records for the same `node_id`.
+
+> Note: If the number of ENR records cannot be encoded into a single message, then they should be sent back using multiple messages, with the `total` field representing the total number of messages that are being sent.
+
+### Find Content (0x05)
+
+Request either the data payload for a specific piece of content on the network, **or** ENR records of nodes that are closer to the requested content.
+
+```
+message_id := 0x05
+type       := request
+sedes      := Container(content_key: ByteList)
+```
+
+* `content_key`: The key for the content being requested. The encoding of `content_key` is dependant on the network.
+
+### Found Content (0x06)
+
+Response message to Find Content (0x05).
+
+This message can contain either a uTP connection ID, a list of ENRs or the
+requested content.
+
+```
+message_id := 0x06
+type       := response
+sedes      := Union[connection-id: Bytes2, content: Bytes, enrs: List[ByteList, 32]]
+```
+
+* `connection_id`: Connection ID to set up a uTP stream to transmit the requested data.
+    * Connection ID values should be randomly generated.
+* `enrs`: List of bytestrings, each of which is an RLP encoded ENR record.
+    * Individual ENR records **MUST** be closer to the requested content than the responding node.
+    * It is invalid to return multiple ENR records for the same `node_id`.
+* `content`: bytestring of the requested content.
+    * This field **MUST** be used when the requested data can fit in this single response.
+
+If the node does not hold the requested content, and the node does not know of any nodes with eligible ENR values, then the node should return `enrs` as an empty list.
+
+
+### Offer (0x07)
+
+Offer a set of content keys that this node has proofs available for.
+
+```
+message_id := 0x07
+type       := request
+sedes      := Container(content_keys: List[ByteList, max_length=64])
+```
+
+* `content_keys`: A list of encoded `content_key` entries. The encoding of each `content_key` is dependant on the network.
+
+
+### Accept (0x08)
+
+Response message to Offer (0x07).
+
+Signals interest in receiving the offered data fro the corresponding Offer message.
+
+
+```
+message_id := 8
+type       := response
+sedes      := Container(connection_id: Bytes2, content_keys: BitList[max_length=64]]
+```
+
+* `connection_id`: Connection ID to set up a uTP stream to transmit the requested data.
+    * ConnectionID values should be randomly generated.
+* `content_keys`: Signals which content keys are desired.
+    * A bit-list corresponding to the offered keys with the bits in the positions of the desired keys set to `1`.
+
+Upon *sending* this message, the requesting node should *listen* for an incoming uTP stream with the generated `connection_id`.
+
+Upon *receiving* this message, the serving node should initiate a uTP stream.
```

Comment:
I would change it a little bit to  - `the serving node should initiate a uTP stream with the received connection_id`. 

Also maybe this is the good place to define what should happen if the receiver already have opened stream with the received connection_id ?